File: Projects/codechef/ADASHOP2.java
package codechef;

import java.util.Scanner;

class ADASHOP2 {
    private static final int[][] TAIL_STEPS = new int[][]{
            new int[]{2, 2},
            new int[]{3, 1},
            new int[]{1, 3},
            new int[]{2, 4},
            new int[]{1, 5},
            new int[]{5, 1},
            new int[]{6, 2},
            new int[]{7, 1},
            new int[]{1, 7},
            new int[]{2, 8},
            new int[]{8, 2},
            new int[]{7, 3},
            new int[]{8, 4},
            new int[]{4, 8},
            new int[]{5, 7},
            new int[]{6, 8},
            new int[]{8, 6},
            new int[]{7, 7},
            new int[]{8, 8}
    };

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int t = stdin.nextInt();
        while (t-- > 0) {
            int r0 = stdin.nextInt();
            int c0 = stdin.nextInt();
            if (r0 != 1 || c0 != 1) {
                System.out.println(2 + TAIL_STEPS.length);
                System.out.printf("%d %d\n", (r0 + c0) / 2, (r0 + c0) / 2);
                System.out.printf("%d %d\n", 1, 1);
            } else {
                System.out.println(TAIL_STEPS.length);
            }
            for (int[] tailStep : TAIL_STEPS) {
                System.out.printf("%d %d\n", tailStep[0], tailStep[1]);
            }
        }
        stdin.close();
    }
}

File: Projects/codechef/ADASHOP2.kt
package codechef

import kotlin.Throws
import codechef.KNPSK.Record

internal object ADASHOP2 {
    private val TAIL_STEPS = arrayOf(intArrayOf(2, 2), intArrayOf(3, 1), intArrayOf(1, 3), intArrayOf(2, 4), intArrayOf(1, 5), intArrayOf(5, 1), intArrayOf(6, 2), intArrayOf(7, 1), intArrayOf(1, 7), intArrayOf(2, 8), intArrayOf(8, 2), intArrayOf(7, 3), intArrayOf(8, 4), intArrayOf(4, 8), intArrayOf(5, 7), intArrayOf(6, 8), intArrayOf(8, 6), intArrayOf(7, 7), intArrayOf(8, 8))
    fun main(args: Array<String?>?) {
        val reader = Scanner(System.`in`)
        var t: Int = reader.nextInt()
        while (t-- > 0) {
            val r0: Int = reader.nextInt()
            val c0: Int = reader.nextInt()
            if (r0 != 1 || c0 != 1) {
                System.out.println(2 + TAIL_STEPS.size)
                System.out.printf("%d %d\n", (r0 + c0) / 2, (r0 + c0) / 2)
                System.out.printf("%d %d\n", 1, 1)
            } else {
                System.out.println(TAIL_STEPS.size)
            }
            for (i in TAIL_STEPS.indices) {
                System.out.printf("%d %d\n", TAIL_STEPS[i][0], TAIL_STEPS[i][1])
            }
        }
        reader.close()
    }
}